File: pascal.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
      
      //TC:O(mn) SC: O(n)
        List<List<Integer>> result =new ArrayList<>();
        
         for(int row=0;row<numRows;row++){
            List<Integer> temp=new ArrayList<>();
             for(int col=0;col<=row;col++){
              if(col==0 ||col==row)
                temp.add(1);
                 else{
                     temp.add(result.get(row-1).get(col-1)+result.get(row-1).get(col));
                }
             }
             result.add(temp);
         }
        
        return result;
    }
}

File: pascal.cpp
TC: O(numRows^2); SC:O(1)
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>>ans;
        ans.push_back({1});
        if(numRows==1)
        return ans;

    
        ans.push_back({1,1});

        for(int i=2;i<numRows;i++)
        {
            vector<int>temp;
            temp.push_back(1);

            for(int j=0;j<ans[i-1].size()-1;j++)
            {
                temp.push_back(ans[i-1][j]+ans[i-1][j+1]);
            }
            temp.push_back(1);
            ans.push_back(temp);
        }

        return ans;
    }
};